### ISSUE TITLE
dask.array.indices incorrectly sets up chunks for each dimension

### ISSUE THREAD

**Issue by sjperkins** [dask/dask#3166]
I ran into some strange behaviour when running `da.nonzero` on the result of `da.ediff1d`.
When one uses the `to_begin=1` kwarg of ediff1d, a chunk of length 1 is added to the front of the resulting dask array. This seems to cause `da.nonzero` to enter into an infinite loop.

## Software Versions

- python 2.7.12
- dask 0.17.0

## Example

```python
from __future__ import print_function

import dask
import dask.array as da

if __name__ == "__main__":
    N = 975409
    chunks = 100000

    A = da.linspace(1.4, 1.5, N, chunks=chunks)
    # Evaluates
    print(da.nonzero(da.ediff1d(A))[0].compute())
    # Evaluates
    print(da.nonzero(da.ediff1d(A, to_end=1))[0].compute())
    # Gets stuck in a recursive loop
    print(da.nonzero(da.ediff1d(A, to_begin=1, to_end=1))[0].compute())
```

**Comment by sjperkins** [comment]
Suspect the problem occurs [here](https://github.com/dask/dask/blob/512aaec69b52be0461046f2973b58240784d748f/dask/array/creation.py#L378-L380) in `dask/creation.py`

```
    dimensions = tuple(dimensions)
    dtype = np.dtype(dtype)
    chunks = tuple(chunks)


    if len(dimensions) != len(chunks):
        raise ValueError("Need same number of chunks as dimensions.")


    xi = []
    for i in range(len(dimensions)):
        xi.append(arange(dimensions[i], dtype=dtype, chunks=chunks[i]))
```

The arange command produces a dask array consisting of entirely of chunks of size 1

**Comment by sjperkins** [comment]
Probably a quick fix, working on a PR now.

**Comment by sjperkins** [comment]
Fixed by #3167
